### PR TITLE
[RHELC-781, RHELC-821] Don't require --pool or --auto-attach 

### PR DESCRIPTION
**Pull request by pr-watson**
This PR removes the need for --pool or --auto-attach options to be required. The listing of available subscriptions to the user has been removed as well as the code hacks introduced to make it possible - these removals are apart of a separate task [RHELC-821]. If SCA is enabled we don't attach any subscription as SCA handles the attachment. A warning has been added if SCA is enabled and the --pool option is added as a CLI argument. If SCA is not enabled and neither the --pool or --auto_attach options are not specified, the code defaults to auto attaching subscriptions similar to how subscription-manager functions. In the event the subscription attachment fails, an error message is printed to the user pointing them to https://access.redhat.com/management/ and mentioning that there they can either enable the SCA, or create an activation key, or find a Pool ID of the subscription they wish to use and pass it to convert2rhel through the --pool CLI option. 

Jira Issues: [RHELC-781](https://issues.redhat.com/browse/RHELC-781), [RHELC-821](https://issues.redhat.com/browse/RHELC-821)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [x] PR title explains the change from the user's point of view
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

**Comment by danmyway** [comment]
The process is not getting killed by `ctrl+c`, I would wait for #759 to get merged, which will resolve these issues

File: tests/integration/tier0/rollback-handling/test_rollback_handling.py
Comment (kokesak) [inline tests/integration/tier0/rollback-handling/test_rollback_handling.py:163]:
```suggestion
    Verify that the subscription is auto-attached.
```

**Review by kokesak — CHANGES_REQUESTED** [review]
The QE part reviewed

File: convert2rhel/subscription.py
Comment (r0x0d) [inline convert2rhel/subscription.py:676]: I think we should compare the output here in lower case, right? In case that they change the casing in the future, it will still work as it will check for the lower-case version of that output always. 

```suggestion
    if "content access mode is set to simple content access." in output.lower():
```

File: tests/integration/tier0/user-prompt-response/test_user_prompt_response.py
Comment (danmyway) [reply tests/integration/tier0/user-prompt-response/test_user_prompt_response.py:19]: Tried to make the comments better overall.

**Review by r0x0d — CHANGES_REQUESTED** [review]
In a general way, looks very good! Thanks for the effort, guys! 

Just a few comments and changes requested, not sure if all of them can be applied in such a short time as we have right now (don't worry if  we can't apply then).

File: tests/integration/tier0/rollback-handling/test_rollback_handling.py
Comment (kokesak) [inline tests/integration/tier0/rollback-handling/test_rollback_handling.py:189]: I would add `timeout` here as well. Since it is everywhere above. Or is there a reason not to have timeout here?

File: tests/integration/tier0/rollback-handling/test_rollback_handling.py
Comment (kokesak) [inline tests/integration/tier0/rollback-handling/test_rollback_handling.py:200]: The same with adding `timeout` here.

File: tests/integration/tier0/user-prompt-response/test_user_prompt_response.py
Comment (kokesak) [inline tests/integration/tier0/user-prompt-response/test_user_prompt_response.py:14]: Missing timeouts here as well. I would personally add them into this files as well, but leave it up to you. Just please stay consistent (either use them everywhere or nowhere (in the scope of this file).

**Review by kokesak — COMMENTED** [review]
Just noticed missing `timeout` parameters in some function calls

**Review by kokesak — APPROVED** [review]

File: convert2rhel/subscription.py
Comment (pr-watson) [reply convert2rhel/subscription.py:677]: sounds good

**Review by r0x0d — APPROVED** [review]

**Pull request merged**
